blocks/footer/footer.js

```diff
@@ -1,21 +1,155 @@
+    }
+  }
+  if (element.children.length > 0) {
+    for (let i = 0; i < element.children.length; i += 1) {
+      traverseAndPrint(element.children[i], level + 1, columnDiv);
+    }
+  }
+}
+
+function decorateFooterLinks(block) {
+  const footerTop = block.querySelector('.footer-links');
+  const defaultWrapper = footerTop.querySelector('.default-content-wrapper');
+  const row = document.createElement('div');
+  row.classList.add('row');
+  const children = defaultWrapper.children[0];
+  defaultWrapper.innerHTML = '';
+  let index = 0;
+  while (index < children.children.length) {
+    const columnDiv = document.createElement('div');
+    columnDiv.classList.add('links-column');
+    traverseAndPrint(children.children[index], 1, columnDiv);
+    index += 1;
+    row.appendChild(columnDiv);
+  }
+  defaultWrapper.appendChild(row);
+}
+
+function decorateDisclaimer(block) {
+  const disclaimer = block.querySelector('.footer-disclaimer');
+  const defaultWrapper = disclaimer.querySelector('.default-content-wrapper');
+  const children = [...disclaimer.querySelector('.default-content-wrapper').children];
+  defaultWrapper.innerHTML = '';
+
+  const headerDiv = document.createElement('div');
+  headerDiv.classList.add('panel-header');
+  const icon = document.createElement('icon');
+  icon.classList.add('toggle-icon');
+  icon.onclick = onClick;
+  icon.textContent = '+';
+  const disclaimerHeading = document.createElement('strong');
+  disclaimerHeading.textContent = children[0].textContent;
+  children[0].textContent = ' ';
+  const panelBody = document.createElement('div');
+  panelBody.classList.add('panel-body');
+  panelBody.classList.add('slide-up');
+  children.forEach((childElement) => {
+    if (childElement.tagName === 'H3') {
+      const span = document.createElement('span');
+      span.textContent = childElement.textContent;
+      const paragraph = document.createElement('p');
+      paragraph.appendChild(span);
+      panelBody.appendChild(paragraph);
+    } else {
+      panelBody.appendChild(childElement);
+    }
+  });
+  headerDiv.appendChild(disclaimerHeading);
+  headerDiv.appendChild(icon);
+  defaultWrapper.appendChild(headerDiv);
+  defaultWrapper.appendChild(panelBody);
+}
 
 /**
  * loads and decorates the footer
  * @param {Element} block The footer block element
  */
 export default async function decorate(block) {
-  const footerMeta = getMetadata('footer');
   block.textContent = '';
-
-  // load footer fragment
-  const footerPath = footerMeta.footer || '/footer';
-  const fragment = await loadFragment(footerPath);
-
-  // decorate footer DOM
-  const footer = document.createElement('div');
-  while (fragment.firstElementChild) footer.append(fragment.firstElementChild);
-
-  block.append(footer);
+  // fetch footer content
+  const footerPath = '/footer';
+  const resp = await fetch(`${footerPath}.plain.html`, window.location.pathname.endsWith('/footer') ? { cache: 'reload' } : {});
+  if (resp.ok) {
```

Comment:
You don't need to fetch footer's markup. It should already be there in the `block`.

Reply:
we are not using "footer" as block in doc , here auto-blocking is loading footer block in which we are deciding that which doc is rendered as footer.

Reply:
I think we should use the existing pattern here to load the footer block 
```
  const footerPath = footerMeta.footer || '/footer';
  const fragment = await loadFragment(footerPath);
```
Once this is loaded you will be able to decorate it further in the code

Reply:
@vivesing thx , it was oversight. Done